src/dave_data/datapool/hotmaps/hotmaps_request.py

```diff
@@ -0,0 +1,337 @@
+
+            # Update the metadata
+            out_meta.update(
+                {
+                    "driver": "GTiff",
+                    "height": out_image.shape[1],
+                    "width": out_image.shape[2],
+                    "transform": out_transform,
+                }
+            )
+        except Exception as e:
+            print(e.__class__.__name__)
+
+    # Save the clipped GeoTIFF
+    with rasterio.open(clipped_raster_path, "w", **out_meta) as dest:
+        dest.write(out_image)
+    return clipped_raster_path
+
+
+def raster_to_vector(output_directory, title, epsg):
+    # Define the path to save the clipped raster file
+    clipped_raster_path = os.path.join(output_directory, title + ".tif")
+    # Define the path to save the vector file in gpkg format
+    vector_path = os.path.join(output_directory, title + ".gpkg")
+
+    # Check if clipped_raster_path is a string and the file exists
+    if (
+        not isinstance(clipped_raster_path, str)
+        or not Path(clipped_raster_path).exists()
+    ):
+        raise FileNotFoundError(
+            f"Raster file '{clipped_raster_path}' does not exist or is not a valid path."
+        )
+
+    # Check if vector_path is a string
+    if not isinstance(vector_path, str):
+        raise ValueError(f"Vector path '{vector_path}' is not a valid string.")
+
+    # Check if epsg is a valid integer
+    if not isinstance(epsg, int):
+        raise ValueError(f"EPSG code '{epsg}' is not a valid integer.")
+
+    with rasterio.open(clipped_raster_path) as src:
+        # Read the first band of the raster image
+        image = src.read(1)
+        # Create a mask for non-zero values
+        mask = image != 0
+
+        # Extracting the CRS from the raster
+        raster_crs = src.crs
+        raster_transform = src.transform
+
+        # Generate shapes (polygons) from the raster
+        results = (
+            {"properties": {"value": value}, "geometry": geometry}
+            for geometry, value in shapes(
+                image, mask=mask, transform=raster_transform
+            )
+        )
+
+        # Collect the features
+        features = list(results)
+
+        # Convert the extracted shapes to a GeoDataFrame
+        geometries = [shape(feature["geometry"]) for feature in features]
+        values = [feature["properties"]["value"] for feature in features]
+        vector = gpd.GeoDataFrame(
+            {"geometry": geometries, "value": values}, crs=raster_crs
+        )
+        # Reproject the vector to the EPSG of interest
+        vector_proj = vector.to_crs(epsg=epsg)
+
+        # Save the Geo-dataframe as a Geopackage
+        vector_proj.to_file(vector_path, driver="GPKG")
+        print(
+            f"Vector file '{vector_path}' corresponding to the georeferenced raster file is "
+            f"generated and added to the hotmaps_output folder."
+        )
+        return vector_path
+
+
+def hotmaps_request(layer_name, bbox, epsg):
```

Comment:
Would be better to input a polygon object because this should be the basis in dave_data. Within the function you can transform the polygon into a bounding box, maybe write a function for this transformation.